File: Exercism Codes/Java/DiamondPrinter.java
import java.util.*;
import java.util.stream.*;

class DiamondPrinter {

    public List<String> printToList(char letter) {
        int size = 2 * (letter - 'A') + 1;
        return reflect(IntStream.range(0, size / 2 + 1).mapToObj(idx -> {
            char c = (char) ('A' + idx);
            char[] buffer = " ".repeat(size).toCharArray();
            int leftPosition = size / 2 + 1 - idx - 1;
            buffer[leftPosition] = c;
            buffer[size - leftPosition - 1] = c;
            return new String(buffer);
        }).collect(Collectors.toCollection(ArrayList::new)));
    }

    private List<String> reflect(List<String> lst) {
        lst.addAll(IntStream.range(1, lst.size()).mapToObj(i -> lst.get(lst.size() - i - 1)).toList());
        return lst;
    }

}

File: Exercism Codes/Kotlin/DiamondPrinter.kt
import kotlin.math.min

class DiamondPrinter {

    fun printToList(letter: Char): List<String> {
        val bufferLength = ((letter - 'A') * 2) + 1
        return List(bufferLength) { idx ->
            val buffer = MutableList(bufferLength) { ' ' }
            val currIdx = min(idx, (bufferLength - 1) - idx)
            buffer[(bufferLength / 2) - currIdx] = 'A' + currIdx
            buffer[(bufferLength / 2) + currIdx] = 'A' + currIdx
            buffer.joinToString("")
        }
    }
}
